core/src/main/java/jenkins/security/MasterToSlaveCallable.java

```diff
@@ -11,10 +17,31 @@
  * @since 1.587 / 1.580.1
  */
 public abstract class MasterToSlaveCallable<V, T extends Throwable> implements Callable<V,T> {
+
+    private static final long serialVersionUID = 1L;
+
     @Override
     public void checkRoles(RoleChecker checker) throws SecurityException {
         checker.check(this,Roles.SLAVE);
     }
 
-    private static final long serialVersionUID = 1L;
+    //TODO: replace by Callable#getChannelOrFail() once Minimal supported Remoting version is 3.15 or above
+    @Restricted(NoExternalUse.class)
```

Comment:
I think you could do something like

```java
@Override
public Channel getChannelOrFail() throws ChannelStateException {
    // copy implementation from Callable
}
```

which would avoid polluting the API with even a restricted function, while still fixing the problem.

Reply:
I was not sure it's 100% safe for defaults and decided not to try due to the urgency. Can try it now since I have a test which should reveal the problem if any

Reply:
Not at all urgent, just an idea which might simplify the patch.